google-cloud-apis/src/com/google/cloud/tools/intellij/cloudapis/CloudApiActionDecoratorExtension.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.cloudapis;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.extensions.ExtensionPointName;
+
+/**
+ * An extension point to dynamically change behaivor of "Add Cloud APIs" action, i.e. check for
+ * eligibility of specific build systems or frameworks and disable/enable/modify text based on that.
+ */
+public interface CloudApiActionDecoratorExtension {
```

Comment:
I may have missed this, but shouldn't there be a Maven version of this extension to check, from a maven perspective, if the action is supported (`MavenCloudApiActionDecoratorExtension` or something)?

Reply:
We essentially do the Maven check now using the IDE dependency system - since the Maven API module strictly depends on presence of Maven we can be sure Maven is present. I see your point though that the project itself (or selected module) is not guaranteed to be Maven one. In this case nothing will be done with POM since the module is not "mavenized" but the service account will be created, and environment variables set into existing configurations.
I think we choose from two things here:
* leave this for now and add a help message in the next PR that will show that the following libraries need to be added into the project (i.e. library name and version) or start using Maven and then re-run this action. Similar thing can be added for Gradle projects.
* hide Maven UI and library version to avoid confusion in this case.

Reply:
> We essentially do the Maven check now using the IDE dependency system - since the Maven API module strictly depends on presence of Maven we can be sure Maven is present. I see your point though that the project itself (or selected module) is not guaranteed to be Maven one. In this case nothing will be done with POM since the module is not "mavenized" but the service account will be created, and environment variables set into existing configurations.

Just trying to understand this. I think there are two separate concepts like you are saying:

1) Is Maven available (i.e. the user has the Maven plugin installed)
2) Is the project/module itself a Maven project

I know we previously were disabling the action when #2 is not true. But now the idea is that we want to enable it for all projects so its not tied to Maven right? I'm fine leaving it in its current state (just do nothing to the project when its not Mavenized) and exploring how to improve this in a separate PR (e.g. making the wording more about the API management / Service account generation instead of build manipulation).

Reply:
You're absolutely correct on those two points. Now, after this PR, Cloud APIs are available not only for all IDEs but also for all Java projects (previously Maven only) which is definitely a better way for users to get access to APIs. We can discuss together how we should present that we support Maven builds and make it clear with wording, just like you said.